src/partition.c

```diff
@@ -1387,7 +1732,7 @@ void partition_gather_weights(void *map_data, int num_elements,
 
   double *weights_e = mydata->weights_e;
   double *weights_v = mydata->weights_v;
-  idx_t *inds = mydata->inds;
+  int *inds = mydata->inds;
```

Comment:
As with the over `IDX` related types these can be 64 bit, so we'll need to take more care about this.

Reply:
Just checking I see there is a  type `SCOTCH_Idx` and define `SCOTCH_NUMMAX`. Probably suitable for sorting
out a unified type.

Reply:
Reverted back to `idx_t` which is now defined for Scotch [case](https://github.com/UCL/swiftsim/blob/cb06b0e5c3d8457c474d0084d973f437d29b20d8/src/partition.c#L71-L76)